rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -305,10 +307,14 @@ void SequentialWriter::write(std::shared_ptr<rosbag2_storage::SerializedBagMessa
   const auto message_timestamp = std::chrono::time_point<std::chrono::high_resolution_clock>(
     std::chrono::nanoseconds(message->time_stamp));
 
-  if (should_split_bagfile(message_timestamp)) {
-    split_bagfile();
+  if (is_first_message_) {
```

Comment:
`is_first_message_` flag already exists (came in with https://github.com/ros2/rosbag2/pull/1037), so we will not break ABI compatibility at all.

`colcon test --event-handlers console_direct+ --packages-select rosbag2_cpp` test all passes.
i verified that [original issue](https://github.com/ros2/rosbag2/issues/966) still exists in humble, but this patch can resolve that.